src/Engine/Data/DrawPrimitives.cpp

```diff
@@ -244,7 +244,7 @@ LineMeshPtr CircleArc( const Core::Vector3& center,
 }
 
 MeshPtr Sphere( const Core::Vector3& center, Scalar radius, const Core::Utils::Color& color ) {
-    auto geom   = makeGeodesicSphere( radius, 2 );
+    auto geom   = makeGeodesicSphere( radius, 4, color );
```

Comment:
why "4". Should'nt it be a typed constant or a function parameter?

Reply:
why not, that's from previous code. 
Anyway I think DrawPrimitives specific code should disapear and use Core directly.
Core function already takes parameters or are templated with such constant.